yaml.go

```diff
@@ -134,33 +134,51 @@ func mergeMaps(dst interface{}, src interface{}) (interface{}, error) {
 // file names. All the objects are going to be merged and arrays/values
 // overridden in the order of the files.
 func NewYAMLProviderFromFiles(files ...string) (Provider, error) {
-	readers, err := filesToReaders(files...)
+	readclosers, err := filesToReaders(files...)
 	if err != nil {
 		return nil, err
 	}
 
-	p, err := newYAMLProviderCore(readers...)
-	if err != nil {
-		return nil, err
+	readers := make([]io.Reader, len(readclosers))
+	for i, r := range readclosers {
+		readers[i] = r
 	}
 
-	return newCachedProvider(p)
+	provider, err := newYAMLProviderFromReader(readers...)
+
+	for _, r := range readclosers {
+		r.Close()
```

Comment:
this [Close()](https://golang.org/pkg/io/#Closer) produces an error, which you skipped.

Reply:
I'm not sure what error you expect to receive here.  If the files were opened for writing, then sure, the error is important since it could indicate that the write failed, but for reading?  The important error is the one returned by newYAMLProviderFromReader() above.  If that function returns an error, then I would think that would be the most important error to return, and the rest can be ignored.  If it does not return an error, meaning that all files were completely read and parsed successfully, then what value is there in returning any error from these calls to Close(), which I still can't think of how that would happen.

Even so, I'd be inclined to return the first error, and not try to concatenate all of the errors that occurred.  So something like:

    for _, r := range readclosers {
        e := r.Close()
        if err == nil {
            err = e
        }

Reply:
Well, the most common case for errors from Close is that the file is already closed, which can't happen here, but there is also a case when the system call returned an error(EINTR). Which probably never happens in practice, but I still prefer to just to handle it and forget(how - doesn't really matter, aggregation is the simplest thing that came to my mind)

Reply:
It's good practice to handle the error unless you have a very specific reason for ignoring the error. Plus the contract for `io.Closer` doesn't let you make any assumptions about the error.

You can use [go.uber.org/multierr](https://godoc.org/go.uber.org/multierr) to do the concatenation for you without loss of information.

```go
var err error
for _, r := range readclosers {
  err = multierr.Append(err, r.Close())
}
```